src/cheri/riscv-unpriv-integration.adoc

```diff
@@ -375,6 +389,42 @@ The zero register is extended with zero metadata and a cleared valid tag: this i
 
 NOTE: Instruction encodings refer to *c* registers when they use `cs1`, `cd`, etc. instead of `rs1`, `rd` (which reads/writes the XLEN-bit subset of the register).
 
+==== Added CSRs
+
+{cheri_base32_ext_name} and {cheri_base64_ext_name} add the capability CSR shown in
+xref:utidc-csrnames-added[xrefstyle=short].
+
+[[utidc-csrnames-added]]
+.Unprivileged capability CSRs added in {cheri_base32_ext_name} and {cheri_base64_ext_name}
+[%autowidth,float="center",align="center",cols="<,<,<,<",options="header"]
+|===
+|CLEN CSR|Address|Permissions|Description
+|<<utidc>>|0x480|RW, <<asr_perm>> required for writes, not reads|User Default Data Capability
+|===
+
+[#utidc,reftext="utidc"]
+===== User Thread Identifier Capability (utidc)
+
+The <<utidc>> register is used to identify the current software thread in user mode.
+On reset the valid tag of <<utidc>> will be set to 0 and the remainder of the data is UNSPECIFIED.
+
+
+NOTE: Compartmentalization seeks to separate the privileges between different
+protection units, e.g., two or more libraries. Code can be separated by sentries,
+which allow for giving out code capabilities to untrusted code where the untrusted
+code can only call the code capability, but not modify it. Sentries can
+be called from different software threads and thus there needs to be a way of
+identifying the current software thread. While identifying the current software thread
```

Comment:
why is there a need to identify the software thread, and what is done with that information?

I think we need a bit more usage information here